fe/fe-core/src/main/java/com/starrocks/catalog/HiveTable.java

```diff
@@ -330,25 +325,23 @@ private void validate(Map<String, String> properties) throws DdlException {
             LOG.warn("property " + HIVE_METASTORE_URIS + " will be ignored " +
                     "and hive table will be created by using property " + HIVE_RESOURCE + " only.");
         }
-
         copiedProps.remove(HIVE_RESOURCE);
-        Resource resource = GlobalStateMgr.getCurrentState().getResourceMgr().getResource(resourceName);
-        if (resource == null) {
-            throw new DdlException("hive resource [" + resourceName + "] not exists");
-        }
-        if (resource.getType() != ResourceType.HIVE) {
-            throw new DdlException("resource [" + resourceName + "] is not hive resource");
+
+        if (isInternalCatalog(resourceName)) {
+            checkResource(resourceName);
         }
-        HiveResource hiveResource = (HiveResource) resource;
-        hiveProperties.put(HIVE_METASTORE_URIS, hiveResource.getHiveMetastoreURIs());
+
         this.resourceName = resourceName;
 
         // check column
         // 1. check column exists in hive table
         // 2. check column type mapping
         // 3. check hive partition column exists in table column list
-        org.apache.hadoop.hive.metastore.api.Table hiveTable = GlobalStateMgr.getCurrentState().getHiveRepository()
-                .getTable(resourceName, this.hiveDb, this.hiveTable);
+        org.apache.hadoop.hive.metastore.api.Table hiveTable = hmsTable;
```

Comment:
can be removed

Reply:
It's necessary to getTable when creating hive external table

Reply:
done